qcodes/instrument_drivers/signal_hound/USB_SA124B.py

```diff
@@ -1,166 +1,327 @@
                            initial_value=5e9,
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Numbers())
+                           vals=vals.Numbers(),
+                           parameter_class=SweepTraceParameter)
         self.add_parameter('span',
                            label='Span ',
                            unit='Hz',
                            initial_value=.25e6,
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Numbers())
+                           vals=vals.Numbers(),
+                           parameter_class=SweepTraceParameter)
+        self.add_parameter('npts',
+                           label='Number of Points',
+                           get_cmd=None,
+                           set_cmd=False)
+        self.add_parameter('avg',
+                           label='Averages',
+                           initial_value=1,
+                           get_cmd=None,
+                           set_cmd=None,
+                           vals=vals.Ints())
         self.add_parameter('power',
                            label='Power ',
                            unit='dBm',
-                           initial_value=0,
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Numbers(max_value=20))
+                           get_cmd=self._get_power_at_freq,
+                           set_cmd=False)
         self.add_parameter('ref_lvl',
                            label='Reference power ',
                            unit='dBm',
                            initial_value=0,
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Numbers(max_value=20))
+                           vals=vals.Numbers(max_value=20),
+                           parameter_class=TraceParameter)
         self.add_parameter('external_reference',
-                           get_cmd=None, set_cmd=None,
                            initial_value=False,
-                           vals=vals.Bool())
+                           vals=vals.Bool(),
+                           parameter_class=TraceParameter)
         self.add_parameter('device_type',
-                           get_cmd=self._do_get_device_type)
-
+                           set_cmd=False,
+                           get_cmd=self._get_device_type)
         self.add_parameter('device_mode',
-                           initial_value='sweeping',
-                           get_cmd=None, set_cmd=None,
-                           vals=vals.Anything())
+                           get_cmd=lambda: 'sweeping',
+                           set_cmd=False)
         self.add_parameter('acquisition_mode',
-                           get_cmd=None, set_cmd=None,
-                           initial_value='average',
-                           vals=vals.Enum('average', 'min-max'))
+                           get_cmd=lambda: 'average',
```

Comment:
I believe @nataliejpg removed it because, It never worked (returned the same array twice due to a typo), It's not that useful as it will only return 2 different values under specific circumstances where 2 measurements overlap